clients/admin-ui/src/features/data-discovery-and-detection/action-center/utils/useAggregateColumnsUtils.tsx

```diff
@@ -0,0 +1,291 @@
+    render: (_, record) => (
+      <DiscoveredSystemStatusCell system={record} rowClickUrl={rowClickUrl} />
+    ),
+  },
+  {
+    title: "Type",
+    dataIndex: "metadata",
+    key: "app_type",
+    render: (metadata: OktaAppMetadata) => (
+      <span>{metadata?.app_type || "-"}</span>
+    ),
+  },
+  {
+    title: "Status",
+    dataIndex: "metadata",
+    key: "status",
+    render: (metadata: OktaAppMetadata) => (
+      <span>{metadata?.status || "-"}</span>
+    ),
+  },
+  {
+    title: "Vendor",
+    dataIndex: "vendor_id",
+    key: "vendor",
+    render: (
+      vendorId: string,
+      record: SystemStagedResourcesAggregateRecord,
+    ) => (
+      <VendorMatchBadge
+        vendorName={vendorId}
+        vendorLogoUrl={record.metadata?.vendor_logo_url}
+        confidence={record.metadata?.vendor_match_confidence}
+      />
+    ),
+  },
+  {
+    title: "Sign-on URL",
+    dataIndex: "metadata",
+    key: "sign_on_url",
+    render: (metadata: OktaAppMetadata) =>
+      metadata?.sign_on_url ? (
+        <a
+          href={metadata.sign_on_url}
+          target="_blank"
+          rel="noopener noreferrer"
+        >
+          View
+        </a>
+      ) : null,
+  },
+  {
+    title: "Created",
+    dataIndex: "metadata",
+    key: "created",
+    render: (metadata: OktaAppMetadata) => (
+      <span>
+        {metadata?.created
+          ? new Date(metadata.created).toLocaleDateString()
+          : NO_VALUE}
+      </span>
+    ),
+  },
+];
+
+const buildBaseColumns = ({
+  consentStatus,
+  rowClickUrl,
+  isDataUsesExpanded,
+  isLocationsExpanded,
+  isDomainsExpanded,
+  dataUsesVersion,
+  locationsVersion,
+  domainsVersion,
+  setIsDataUsesExpanded,
+  setIsLocationsExpanded,
+  setIsDomainsExpanded,
+  setDataUsesVersion,
+  setLocationsVersion,
+  setDomainsVersion,
+}: Omit<
+  ColumnBuilderParams,
+  "monitorId" | "readonly" | "allowIgnore" | "onTabChange"
+>): ColumnsType<SystemStagedResourcesAggregateRecord> => [
+  {
+    title: () => (
+      <Space>
+        <div>System</div>
+        <DiscoveryStatusIcon consentStatus={consentStatus} />
+      </Space>
+    ),
+    dataIndex: "name",
+    key: DiscoveredSystemAggregateColumnKeys.SYSTEM_NAME,
+    fixed: "left",
+    render: (_, record) => (
+      <DiscoveredSystemStatusCell system={record} rowClickUrl={rowClickUrl} />
+    ),
+  },
+  {
+    title: "Assets",
+    dataIndex: "total_updates",
+    key: DiscoveredSystemAggregateColumnKeys.TOTAL_UPDATES,
+  },
+  {
+    title: "Categories of consent",
+    key: DiscoveredSystemAggregateColumnKeys.DATA_USE,
+    menu: {
+      items: expandCollapseAllMenuItems,
+      onClick: (e) => {
+        e.domEvent.stopPropagation();
+        if (e.key === "expand-all") {
+          setIsDataUsesExpanded(true);
+          setDataUsesVersion((prev) => prev + 1);
+        } else if (e.key === "collapse-all") {
+          setIsDataUsesExpanded(false);
+          setDataUsesVersion((prev) => prev + 1);
+        }
```

Comment:
nit: This code is duplicated on lines 199 and 240. It should be extracted to DRY it up (I know, this is pre-existing code so you can ignore this one)